impeller/renderer/backend/metal/surface_mtl.h

```diff
@@ -65,6 +65,9 @@ class SurfaceMTL final : public Surface {
     present_with_transaction_ = present_with_transaction;
   }
 
+  /// @brief Perform the final blit and trigger end of frame workloads.
+  bool PreparePresent();
```

Comment:
Perhaps `Present` can not be renamed to `SubmitCommandBuffer`? What happens if there is a missing call to prepare present? Does Present do that for you implicitly?

I suppose we can figure this out later.

Reply:
I wil add an FML_CHECK that we've done both. its all our code so we should be able to do that...

Reply:
Yeah, I'm not too concerned. Simple DCHECKS ought to be fine.